core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -260,10 +262,14 @@ protected void doStop() throws Exception {
         throw new ConfigException("Could not parse a listener URI from the `listener` configuration option.");
       }
       String scheme = uri.getScheme();
-      if (scheme != null && (scheme.equals("http") || scheme.equals("https"))) {
+      if (uri.getPort() == -1) {
+        throw new ConfigException("Found a listener without a port. All listeners must have a port. The " +
+          "listener without a port is: " + listenerStr);
+      }
+      if (scheme != null && supportedSchemes.contains(scheme)) {
         listeners.add(uri);
       } else {
-        log.warn("Found a listener with an unsupported scheme (ony http and https are supported). Ignoring " +
+        log.warn("Found a listener with an unsupported scheme (supported: " + supportedSchemes + "). Ignoring " +
```

Comment:
Should we be throwing a ConfigException if this else clause filters out all listeners?